### PR TITLE
feat: session now returns a map to the generated resources

### PR DESCRIPTION
**Pull request by geoand**
Moreover the `SimpleFileWriter` now has the ability to forgo writing the
actual files on the filesystem thus giving the ability for integrations
to control this behavior

### CODE REVIEW

File: core/src/main/java/io/ap4k/SessionWriter.java [hunk 1]
```diff
@@ -23,32 +24,44 @@
...
+    return result;
   }
 
   /**
    * Writes a {@link Configuration}.
    * @param config  The target session configurations.
+   * @return Map Entry containing the file system path of the written configuration and the actual content as the value
    */
-   void write(Configuration config);
+  Map.Entry<String, String> write(Configuration config);
```
Comment (iocanel) [inline core/src/main/java/io/ap4k/SessionWriter.java]: Given that the session itself holds such a map, I don't see much point in also returning it here.

The only differennce I see is that one is a Map<String, KubernetesList> and the other is Map<String ,String>. I'd rather have the Session use a Map<String, String> instead.

File: core/src/main/java/io/ap4k/processor/AptWriter.java [hunk 2]
```diff
@@ -85,7 +81,9 @@ public void write(Configuration config) {
       name = name.toLowerCase();
       FileObject yml = processingEnv.getFiler().createResource(StandardLocation.CLASS_OUTPUT, PACKAGE, getProject().getAp4kOutputDir() + "/" + String.format(CONFIG, name, YML));
       try (Writer writer = yml.openWriter()) {
-        writer.write(Serialization.asYaml(config));
+        final String value = Serialization.asYaml(config);
+        writer.write(value);
+        return new AbstractMap.SimpleEntry<>(yml.toString(), value);
```
Comment (iocanel) [inline core/src/main/java/io/ap4k/processor/AptWriter.java]: The call to `toString()` kind of scares me.

I think that this is more acurate:
    
    Urls.toFile(yml.toUri().toURL()).getAbsolutePath();

Comment (iocanel) [reply core/src/main/java/io/ap4k/processor/AptWriter.java]: Also the javadoc should mention that we are talking about absolute path (we do talk about absolute path, don't we)?

Comment (geoand) [reply core/src/main/java/io/ap4k/processor/AptWriter.java]: Yes, that is correct, it's the absolute path

**Pull request merged**